src/components/authorization/authorization.service.ts

```diff
@@ -265,13 +271,22 @@ export class AuthorizationService {
     }
   }
 
-  isSensitivityAllowed(
+  isSensitivityAllowed<TResource extends ResourceShape<any>>(
     grant: Partial<
       Record<Action, boolean> & Record<'sensitivityAccess', Sensitivity>
     >,
+    resource: TResource,
+    prop: string,
     sensitivity?: Sensitivity
   ): boolean {
     const sensitivityRank = { High: 3, Medium: 2, Low: 1 };
+    if (grant.sensitivityAccess && !sensitivity) {
+      throw new InputException(
```

Comment:
```suggestion
      throw new ServerException(
```
Input is for user input. This would be a developer mistake.